data-collection/src/main/java/refactoringml/App.java

```diff
@@ -118,61 +118,57 @@ public Project run () throws Exception {
 			RevWalk walk = JGitUtils.getReverseWalk(repo, mainBranch);
 			RevCommit currentCommit = walk.next();
 
-			int refactoringMinerTimeout = Integer.valueOf(getProperty("timeoutRefactoringMiner"));
+			int refactoringMinerTimeout = Integer.parseInt(getProperty("timeoutRefactoringMiner"));
 			log.debug("Set Refactoring Miner timeout to " + refactoringMinerTimeout + " seconds.");
-
 			log.info("Start mining project " + gitUrl + "(clone at " + clonePath + ")");
 
 			// we only analyze commits that have one parent or the first commit with 0 parents
 			for (boolean endFound = false; currentCommit!=null && !endFound; currentCommit = walk.next()) {
 				long startCommitTime = System.currentTimeMillis();
 				String commitHash = currentCommit.getId().getName();
 
-				try {
-					db.openSession();
+				// i.e., ignore merge commits
+				if (currentCommit.getParentCount() > 1)
+					continue;
+				// did we find the last commit to process?
+				// if so, process it and then stop
+				if (currentCommit.getName().equals(lastCommitToProcess))
+					endFound = true;
 
-					// i.e., ignore merge commits
-					if (currentCommit.getParentCount() > 1)
-						continue;
-
-					// did we find the last commit to process?
-					// if so, process it and then stop
-					if (currentCommit.equals(lastCommitToProcess))
-						endFound = true;
+				try {
+					// stores the commit meta data
+					CommitMetaData superCommitMetaData = new CommitMetaData(currentCommit, project);
+					//stores all the ck metrics for the current commit
+					List<RefactoringCommit> allRefactoringCommits = new ArrayList<>();
 
-					refactoringsToProcess = null;
-					commitIdToProcess = null;
+					db.openSession();
 					// Note that we only run it if the commit has a parent, i.e, skip the first commit of the repo
-					if (currentCommit.getParentCount() == 1)
+					if (!isFirst(currentCommit)){
```

Comment:
this if can go together with the two other ifs before the try/catch. This way you would avoid having a big nested block

Reply:
We cannot skip the call to: `processMetrics.collectMetrics(currentCommit, superCommitMetaData, allRefactoringCommits);`
Because otherwise all java class files from the first commit are not tracked in the PMDatabase.

Reply:
and is this really bad? The second time the file appears, we will track it. We will only lose a +1 in the commit count and maybe an author (which, in many cases, is just importing all the files to Git).

Reply:
No, it is not really bad, but why should we do it? I can't see the advantages.

Reply:
Only to simplify the code even more, but you already simplified it a lot in this PR, indeed!